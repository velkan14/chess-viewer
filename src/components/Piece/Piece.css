.Piece {
  font-size: 8vh;
  overflow: hidden;
  width: 100%;
  height: 100%;
  color: white;
  cursor: grab;
  -webkit-text-stroke: 0.03rem black;
}

.pieceblack {
  color: #2e2e2e;
  -webkit-text-stroke: 0.03rem white;
}

[draggable="true"] {
  cursor: move;
}

.dragging {
  cursor: grabbing;
  z-index: 2;
}
