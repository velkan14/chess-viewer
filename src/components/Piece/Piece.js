import React, { useState, useRef } from "react";
import classNames from "classnames";
import "./Piece.css";
import Draggable from "react-draggable";

const piecesMapping = {
  p: { char: "♟", color: "black" },
  n: { char: "♞", color: "black" },
  b: { char: "♝", color: "black" },
  r: { char: "♜", color: "black" },
  q: { char: "♚", color: "black" },
  k: { char: "♛", color: "black" },
};

const Piece = (props) => {
  const { piece, position, onDragStop, onDragStart } = props;

  const [dragging, setDragging] = useState(false);
  const nodeRef = useRef(null);

  return (
    <Draggable
      nodeRef={nodeRef}
      position={{ x: 0, y: 0 }}
      onStart={(a) => {
        setDragging(true);
        onDragStart(position);
      }}
      onStop={(e) => {
        setDragging(false);
        onDragStop(e, position);
      }}
    >
      <div
        ref={nodeRef}
        className={classNames("Piece", {
          pieceblack: piece.color === "black",
          dragging: dragging,
        })}
      >
        {piecesMapping[piece.name].char}
      </div>
    </Draggable>
  );
};

export default Piece;
