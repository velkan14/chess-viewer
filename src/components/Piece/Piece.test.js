import React from "react";
import { shallow } from "enzyme";
import shallowToJson from "enzyme-to-json";
import Piece from "./Piece";

test("Test snapshots", () => {
  let wrapper = shallow(
    <Piece
      piece={{ name: "p", color: "black" }}
      position={{ x: 0, y: 0 }}
      onDragStop={() => {}}
      onDragStart={() => {}}
    />
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();

  wrapper = shallow(
    <Piece
      piece={{ name: "k", color: "white" }}
      position={{ x: 0, y: 0 }}
      onDragStop={() => {}}
      onDragStart={() => {}}
    />
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
