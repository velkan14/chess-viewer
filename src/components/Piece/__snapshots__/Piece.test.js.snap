// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshots 1`] = `
<Draggable
  allowAnyClick={false}
  axis="both"
  bounds={false}
  cancel={null}
  defaultClassName="react-draggable"
  defaultClassNameDragged="react-draggable-dragged"
  defaultClassNameDragging="react-draggable-dragging"
  defaultPosition={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  disabled={false}
  enableUserSelectHack={true}
  grid={null}
  handle={null}
  nodeRef={
    Object {
      "current": null,
    }
  }
  offsetParent={null}
  onDrag={[Function]}
  onMouseDown={[Function]}
  onStart={[Function]}
  onStop={[Function]}
  position={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scale={1}
  transform={null}
>
  <div
    className="Piece pieceblack"
  >
    ♟
  </div>
</Draggable>
`;

exports[`Test snapshots 2`] = `
<Draggable
  allowAnyClick={false}
  axis="both"
  bounds={false}
  cancel={null}
  defaultClassName="react-draggable"
  defaultClassNameDragged="react-draggable-dragged"
  defaultClassNameDragging="react-draggable-dragging"
  defaultPosition={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  disabled={false}
  enableUserSelectHack={true}
  grid={null}
  handle={null}
  nodeRef={
    Object {
      "current": null,
    }
  }
  offsetParent={null}
  onDrag={[Function]}
  onMouseDown={[Function]}
  onStart={[Function]}
  onStop={[Function]}
  position={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scale={1}
  transform={null}
>
  <div
    className="Piece"
  >
    ♛
  </div>
</Draggable>
`;
